submissions/Zakkarat/JS-OOP/js/app.js

```diff
@@ -49,49 +67,78 @@ const Player = function(x, y) {
   this.x = x;
   this.y = y;
   this.sprite = 'images/char-boy.png';
+  this.lives = PlayerConf.startLives;
+  this.score = PlayerConf.startScore;
+  this.highScore = PlayerConf.startScore;
 }
 Player.prototype.render = function() {
     ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
 };
 
 Player.prototype.update = function() {
-    if(this.y < 0){
-      this.x = startX;
-      this.y = startY;
-      updScore();
+    if(this.y < Field.maxHeight){
+      this.x = PlayerConf.startX;
+      this.y = PlayerConf.startY;
+      this.updScore();
     }
   }
 
 Player.prototype.handleInput = function(allowedKey) {
     switch (allowedKey){
         case 'left':
-          if(this.x > 0 && allowMove) {
-            this.x -= stepX;
+          if(this.x > Field.minWidth && allowMove) {
+            this.x -= PlayerConf.stepX;
           }
             break;
         case 'up':
           if(allowMove) {
-            this.y -= stepY;
+            this.y -= PlayerConf.stepY;
           }
             break;
         case 'right':
-          if(this.x < 400 && allowMove) {
-            this.x += stepX;
+          if(this.x < Field.maxWidth && allowMove) {
+            this.x += PlayerConf.stepX;
           }
             break;
         case 'down':
-          if(this.y < 370 && allowMove) {
-            this.y += stepY;
+          if(this.y < Field.minHeight && allowMove) {
+            this.y += PlayerConf.stepY;
           }
             break;
     }
 };
+Player.prototype.updLive = function() {
+    this.lives--;
+    if (this.lives == 0){
+      this.lives = PlayerConf.startLives;
+      this.highScr();
+    }
+    scr[0].textContent = this.lives;
+}
+Player.prototype.updScore = function() {
+  this.score++;
+  scr[2].textContent = this.score;
+}
+Player.prototype.highScr = function() {
+  if(this.highScore < this.score) {
+  this.highScore = this.score;
+  scr[1].textContent = this.highScore;
+}
+  this.score = PlayerConf.startScore;
+  scr[2].textContent = this.score;
+  popUp();
+  allowMove = false;
+}
+btn.addEventListener('click', function() {
+  popUp()
+  allowMove = true;
+});
 // Now instantiate your objects.
 // Place all enemy objects in an array called allEnemies
 // Place the player object in a variable called player
-let player = new Player(startX, startY);
+let player = new Player(PlayerConf.startX, PlayerConf.startY);
 
-let allEnemies = [new Enemy(60), new Enemy(140), new Enemy(230)];
+let allEnemies = [new Enemy(EnemyConf.minStartY[0]), new Enemy(EnemyConf.minStartY[1]), new Enemy(EnemyConf.minStartY[2])];
```

Comment:
let allEnemies = minStartY.map(startY => new Enemy(startY)) better I thnik

Reply:
Thank you. Done!